### PR TITLE
Feature/add plots to result

### LINKED ISSUE #254: Adding plots to result file
## Use Case
*Please provide a use case to help us understand your request in context*
we'd like to embed plot visualizations for packings to facilitate a more comprehensive analysis from the simularium result files.

### PR DESCRIPTION
**Pull request by rugeli**
Problem
=======
What is the problem this work solves, including
closes #254 

Solution
========

- created a class to handle plot data(scatter plots and histograms) and added all plots to one list
- in the function `histogram` within `Analysis.py`, added the logic to insert plot data into result file. Set the argument `add_to_result` to true and set `save_png` to false, by default 
- in `simularium_helper`, added plot data to the trajectory converter and had simularium io process the plot display on the website     

with @meganrm @mogres @ascibisz and Joe

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Pack any recipes with the `save_analyze_result` config set to `true` to see if the plots display next to the result as expected 

Screenshots (optional):
-----------------------
Screenshot 2024-05-29 at 1 22 15 PM

### CODE REVIEW

File: cellpack/autopack/Environment.py [hunk 1]
```diff
@@ -143,7 +143,11 @@ def __init__(self, config=None, recipe=None):
         self.name = name
         self.version = recipe.get("version", "default")
         # saving/pickle option
-        self.saveResult = "out" in config
+        self.saveResult = (
+            "out" in config
+            and not config["save_analyze_result"]
```
Comment (mogres) [inline cellpack/autopack/Environment.py:148]: Why do we have these additional conditions here? Wouldn't this prevent saving simularium files completely if there is more than one packing?

**Review by mogres — APPROVED** [review]
Really cool feature! The `PlotData` class is super useful :) I was able to run the `test_spheres` recipe and generate simularium files with plots on the side!

Your PR description asks to set `show_grid_plot` config to `true` but I don't see it checked anywhere. Do you mean `save_analyze_result`?

File: cellpack/autopack/Environment.py (hunk 1 above)
Comment (meganrm) [reply cellpack/autopack/Environment.py:148]: it's because the saving happens from the Analysis class instead of the Env class

**Review by meganrm — APPROVED** [review]

**Comment by rugeli** [comment]
> Really cool feature! The `PlotData` class is super useful :) I was able to run the `test_spheres` recipe and generate simularium files with plots on the side!
> 
> Your PR description asks to set `show_grid_plot` config to `true` but I don't see it checked anywhere. Do you mean `save_analyze_result`?

ah right, I meant `save_analyze_result`, corrected!

**Pull request merged**